### PR TITLE
feat(theming): theming

### PR DESCRIPTION
**Pull request by thomaspink**

### CODE REVIEW

**Comment by thomaspink** [comment]
fixes #58 #59

**Comment by thomaspink** [comment]
fixes #14

File: src/lib/button/button.ts [hunk 1]
```diff
@@ -99,7 +139,7 @@ export class GhButton extends _GhButtonMixinBase implements OnDestroy, CanDisabl
     '[attr.aria-disabled]': 'disabled.toString()',
     '(click)': '_haltDisabledEvents($event)',
   },
-  inputs: ['disabled'],
+  inputs: ['disabled', 'color'],
```
Comment (ffriedl89) [inline src/lib/button/button.ts]: you dont need this color input here right?

File: src/lib/core/style/_interactive-common.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Mixin overriding default focus styles and setting our own
+@mixin gh-focus-style() {
```
Comment (ffriedl89) [inline src/lib/core/style/_interactive-common.scss]: @katrin-freihofner and simon discussed using the native browser focus styles, we should ask them again if we want to use them already

File: src/lib/button/button.ts (hunk 1 above)
Comment (thomaspink) [reply src/lib/button/button.ts]: Yes we need it, as the @Input is defined in the GhButton class and GhAnchorButton does extend

File: src/lib/core/style/_interactive-common.scss (hunk 2 above)
Comment (thomaspink) [reply src/lib/core/style/_interactive-common.scss]: We can change this later if the want to also add it to css-groundhog. i wanna get this branch merged as soon as possible. Maybe add an issue instead?

Comment (ffriedl89) [reply src/lib/core/style/_interactive-common.scss]: i think it's fine for now, since its still up for discussion

**Pull request merged**